info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -261,18 +270,46 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  // @TODO [Find incumbency.]
+  private boolean checkIncumbency(String candidateName, String position, String division) {
+    try {
+      String queryUrl = String.format("%s&ocdDivisionId=%s", REPRENTATIVE_QUERY_URL, division);
+      JsonObject representatives = queryCivicInformation(queryUrl);
+      JsonArray offices = representatives.getAsJsonArray("offices");
+      int i = 0;
```

Comment:
What is the purpose of this counter?

Reply:
There are two JSON arrays: offices and officials. Elements of these two arrays at the same index correspond to each other.
This counter is used to store the index of the office (position) that we are looking for. Then, the official (who's currently in office) would be extracted at the same index stored in the counter, and we check if the name is the same. [This implementation would come right after line 287)